packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -585,29 +505,128 @@ async function fetchDeployments(cloudProvider: CloudProvider['type'], deployment
+    const addressDirectory = getAddressDirectory(directoryStructure, airnodeAddress) as Directory;
+    const stageDirectory = getStageDirectory(directoryStructure, airnodeAddress, stage) as Directory;
+
+    // Delete stage directory and its content
+    logger.debug(`Deleting deployment directory '${stageDirectory.bucketKey}' and its content`);
+    await cloudProviderLib[cloudProviderType].deleteBucketDirectory(bucketName, stageDirectory);
+    // eslint-disable-next-line functional/immutable-data
+    delete addressDirectory.children[stage];
+
+    // Delete Airnode address directory if empty
+    if (Object.keys(addressDirectory.children).length === 0) {
+      logger.debug(`Deleting Airnode address directory '${addressDirectory.bucketKey}'`);
+      await cloudProviderLib[cloudProviderType].deleteBucketDirectory(bucketName, addressDirectory);
+      // eslint-disable-next-line functional/immutable-data
+      delete directoryStructure[airnodeAddress];
+    }
+
+    // Delete the whole bucket if empty
+    if (Object.keys(directoryStructure).length === 0) {
+      logger.debug(`Deleting Airnode bucket '${bucketName}'`);
+      await cloudProviderLib[cloudProviderType].deleteBucket(bucketName);
+    }
+  });
+
+  if (!goRemove.success) {
+    spinner.fail(`Failed to remove Airnode '${deploymentId}'`);
+    throw goRemove.error;
+  }
+
+  spinner.succeed(`Airnode '${deploymentId}' removed successfully`);
+}
+
 export async function listAirnodes(cloudProviders: readonly CloudProvider['type'][]) {
   const deployments: Deployment[] = [];
 
-  for (const cloudProvider of cloudProviders) {
+  for (const cloudProviderType of cloudProviders) {
     // Using different line of text for each cloud provider so we can easily convey which cloud provider failed
     // and which succeeded
     const spinner = logger
       .getSpinner()
-      .start(`Listing Airnode deployments from cloud provider ${cloudProvider.toUpperCase()}`);
+      .start(`Listing Airnode deployments from cloud provider ${cloudProviderType.toUpperCase()}`);
     if (logger.inDebugMode()) {
       spinner.info();
     }
 
-    const goListCloudAirnodes = await go(() => fetchDeployments(cloudProvider));
+    const goListCloudAirnodes = await go(() => fetchDeployments(cloudProviderType));
 
     if (goListCloudAirnodes.success) {
       spinner.succeed();
       deployments.push(...goListCloudAirnodes.data);
     } else {
-      spinner.fail(`Failed to list deployments from ${cloudProvider.toUpperCase()}: ${goListCloudAirnodes.error}`);
+      spinner.fail(`Failed to fetch deployments from ${cloudProviderType.toUpperCase()}: ${goListCloudAirnodes.error}`);
     }
   }
-  const sortedDeployments = sortBy(deployments, ['cloudProvider', 'airnodeAddress', 'stage', 'airnodeVersion']);
+  // Yes, lodash's sortBy can apparently do this
+  // It can access a deeper field and it will skip a nonexistent one as well
+  const sortedDeployments = sortBy(deployments, [
```

Comment:
Does it use the correct compare in the sorting? Not sure if you know, but this:
```js
[2, 10].sort()
```

gives you `[10, 2]`, because the default sorting fn converts the numbers to strings first...

Reply:
Hmm, for the version field, right? I don't know tbh, I'll check it out.